### PR TITLE
Resolve registry related TODOs

### PR DESCRIPTION
**Pull request by LoganHenderson**
Refactor some of the code in the `lens chain` subcommand to resolve the TODO items there.

### CODE REVIEW

**Comment by jackzampolin** [comment]
I would like to keep the previous functionality for chains edit but do like the ability to open the config in an editor easily. How about we put that in `lens editor` or something similar. Also would be nice for users to specify which editor. Defaulting to vim but having the optional ability to pass in another editor program, say `lens editor nano` would be nice.

File: client/chain_client.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ func (cc *ChainClient) PrintObject(res interface{}) error {
 		if err != nil {
 			return err
 		}
-	case "json-indent":
+	case "indent":
```
Comment (jackzampolin) [inline client/chain_client.go:104]: This is nicer

File: cmd/config.go [hunk 2]
```diff
@@ -45,7 +46,8 @@ func createConfig(home string, debug bool) error {
 	return nil
 }
 
-func overwriteConfig(home string, cfg *Config) error {
+func overwriteConfig(cfg *Config) error {
+	home := viper.GetString("home")
```
Comment (jackzampolin) [inline cmd/config.go:50]: Is relying on viper OK here?

**Review by jackzampolin — APPROVED** [review]
This is great, just that one note about the edit command.

**Comment by Zygimantass** [comment]
> I would like to keep the previous functionality for chains edit but do like the ability to open the config in an editor easily. How about we put that in `lens editor` or something similar. Also would be nice for users to specify which editor. Defaulting to vim but having the optional ability to pass in another editor program, say `lens editor nano` would be nice.

Usually commands like `lens editor` would just respect the EDITOR environment variable

File: cmd/config.go (hunk 2 above)
Comment (LoganHenderson) [reply cmd/config.go:50]: It works - if that's what you're asking 😃 . I suppose it's not the cleanest. We could put `home` onto the Config object. 

Seemed like a lot of work to pass in every time. Can revert back to that though if you think it's better.

**Comment by LoganHenderson** [comment]
I picked `LENS_EDITOR` to override the editor, taking inspiration from `KUBE_EDITOR` which is used to override the editor in _kubectl_. I tried printing `EDITOR` on my mac and an Ubuntu container and they returned an empty string - doesn't seem like a standard thing (having that var set).  So, I'm not checking for that too. We could check them both though.

File: cmd/chains.go [hunk 3]
```diff
@@ -242,8 +213,35 @@ func cmdChainsShowDefault() *cobra.Command {
...
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			home, err := cmd.Flags().GetString("home")
+			if err != nil {
+				return err
+			}
+
+			editor := os.Getenv("LENS_EDITOR")
+			if editor == "" {
+				editor = "vim"
```
Comment (prasincs) [inline cmd/chains.go:238]: nit: would be nice to fallback to the system default editor but 🤷 I like vim anyway

**Review by prasincs — APPROVED** [review]

File: cmd/chains.go (hunk 3 above)
Comment (LoganHenderson) [reply cmd/chains.go:238]: I'd love to do this. How do you get the system default editor? I'm not finding anything online that lets you do it easily across Mac/Linux.

Comment (LoganHenderson) [reply cmd/chains.go]: According to [this](https://askubuntu.com/a/709034), the `EDITOR` env var should hold the system default. If it's blank, it uses `vi`. At least for Linux, according to https://askubuntu.com/a/709034

I switched it to check that and default to `vi`. Lmk if you have a better option though.

**Pull request merged**